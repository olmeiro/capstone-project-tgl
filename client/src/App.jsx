import React from 'react'
import { AppRouter } from './routes/AppRouter'
import { HashRouter } from 'react-router-dom'

export default function App () {
  return (
     <HashRouter>
       <AppRouter />
     </HashRouter>
  )
}
