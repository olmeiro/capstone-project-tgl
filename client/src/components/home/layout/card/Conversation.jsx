import { useEffect, useState } from 'react'
import PropTypes from 'prop-types'

import { socialApi } from '../../../../api'
import photoDefault from '../../../../../assets/photoDefault.png'

const Conversation = ({ conversation, currentUser }) => {
  const [friend, setFriend] = useState(null)

  useEffect(() => {
    const friendId = conversation.members.find((id) => id !== currentUser.id)

    const getFriend = async () => {
      const res = await socialApi.get(`/user/byid/${friendId}`)
      const friend = res.data.body
      setFriend(friend)
    }
    getFriend()
  }, [currentUser, conversation])

  return (
    <div className="cursor-pointer relative rounded-lg px-2 py-2 flex items-center space-x-3 hover:border-gray-400 focus-within:ring-2 mb-3 hover:bg-gray-200">
      <div className="flex-shrink-0 ">
        <img
          src={
            friend && friend.photoProfile ? friend.photoProfile : photoDefault
          }
          className="h-10 w-10 rounded-full"
        />
      </div>

      <div className="flex-1 min-w-0">
        <div className="flex items-center justify-between">
          <p className="text-sm font-bold text-team-dark">
            {friend && friend.alias}
          </p>
        </div>
      </div>
    </div>
  )
}
export default Conversation

Conversation.propTypes = {
  conversation: PropTypes.object,
  currentUser: PropTypes.object
}
