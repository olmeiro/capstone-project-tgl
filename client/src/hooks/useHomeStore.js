import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { socialApi } from '../api'
import { checkFavorites, getFavorites, actionKeepLastUserVisitesOnRedux, getPostsToHome, getFriendsFromFriends, checkComments, getUserSearched, checkEmptySearchBar, setPathReference, setLastUserVisited, setInfoFromUserLoggedIn, setcheckInfoUser, setPostsFromUserLoggedIn } from '../store'

export const useHomeStore = () => {
  const dispatch = useDispatch()
  const { user } = useSelector(state => state.auth)
  const userId = user.id

  const entire = socialApi.get(`/user/byid/${userId}`).then(response => response.data.body)
  let friends = entire.then(user => user.friends)

  const getPostsToHomeHook = async () => {
    friends = await friends
    let postsTotalFromFriends = friends.map(async friendId => await socialApi.get(`/posts/byuser/${friendId}`))
    postsTotalFromFriends = Promise.all(postsTotalFromFriends)
    postsTotalFromFriends = await postsTotalFromFriends
    postsTotalFromFriends = postsTotalFromFriends.map(posts => posts.data.body).flat()
    dispatch(getPostsToHome(postsTotalFromFriends))
  }
  const getFriendsFromFriendsHook = async () => {
    const wholeUser = await socialApi.get(`/user/byid/${userId}`)
    const friendsId = wholeUser.data.body.friends
    let suggestion = friendsId.map(async id => {
      const response = await socialApi.get(`/user/byid/${id}`)
      const data = response.data
      const suggestion = data.body.friends
      return suggestion
    })
    suggestion = Promise.all(suggestion)
    suggestion = await suggestion
    suggestion = suggestion.flat()
    suggestion = suggestion.filter((friendid, index) => suggestion.indexOf(friendid) == index)
    suggestion = suggestion.map(async id => {
      const response = await socialApi.get(`/user/byid/${id}`)
      const data = response.data
      const suggestion = data.body
      return suggestion
    })
    suggestion = Promise.all(suggestion)
    suggestion = await suggestion
    dispatch(getFriendsFromFriends(suggestion))
  }
  const addFriendshipHook = async (friendId) => {
    await socialApi.post('/friends/', { userId, friendId })
  }
  const likeAPost = async (id) => {
    await socialApi.put('/posts/', { id, likes: 3 })
  }
  const makeAComment = async (postId, comment) => {
    await socialApi.post('/comments/', { comment, postId, userId })
  }
  const getCommentsByPost = async (postid) => {
    let comments = await socialApi.get(`/comments/bypost/${postid}`)
    comments = comments.data.body
    comments = comments.map(async comment => {
      const userId = comment.UserId
      let user = await socialApi.get(`/user/byid/${userId}`)
      user = user.data.body
      comment.user = user
      return comment
    })
    comments = await Promise.all(comments)
    return comments
  }
  const deleteComment = async (commentId) => {
    await socialApi.delete(`/comments/${commentId}`)
  }
  const checkCommentsHook = () => {
    dispatch(checkComments())
  }
  const searchUserByAlias = async (aliasFromBody) => {
    let foundUsers = null
    if (aliasFromBody.trim() != '') {
      const response = await socialApi.get('/user/all')
      const users = response.data.body
      foundUsers = users.filter(user => user.alias.toLowerCase().includes(aliasFromBody && aliasFromBody.toLowerCase()))
    }
    dispatch(getUserSearched(foundUsers))
  }
  const checkEmptySearchBarHook = () => {
    dispatch(checkEmptySearchBar())
  }
  const sendPathHook = (pathReference) => {
    dispatch(setPathReference(pathReference))
  }
  const setLastUserVisitedHook = async (userAlias, userId) => {
    const responseUser = await socialApi.get(`user/byalias/${userAlias}`)
    const user = responseUser.data.body
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    user.posts = posts
    localStorage.setItem('lastUserVisited', JSON.stringify(user))
    const lastUserVisited = JSON.parse(localStorage.getItem('lastUserVisited'))
    dispatch(setLastUserVisited(lastUserVisited))
  }
  const keepLastUserVisitesOnRedux = async () => {
    const lastUserVisited = JSON.parse(localStorage.getItem('lastUserVisited'))
    dispatch(actionKeepLastUserVisitesOnRedux(lastUserVisited))
  }
  const getInfoFromTheUserLoggedIn = async () => {
    const user = await entire
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    const newUser = {
      ...user,
      posts
    }
    dispatch(setInfoFromUserLoggedIn(newUser))
  }
  const checkInfoUserHook = async () => {
    dispatch(setcheckInfoUser())
  }
  const getPostsFromUserLoggedIn = async () => {
    const responsePosts = await socialApi.get(`/posts/byuser/${userId}`)
    const posts = responsePosts.data.body
    dispatch(setPostsFromUserLoggedIn(posts))
  }
  const addToFavorites = async (postFavoriteId, userLogeadoId) => {
    await socialApi.post(`/favorites?postId=${postFavoriteId}&userId=${userLogeadoId}`)
  }
  const getFavoritesHook = async () => {
    const response = await socialApi.get(`/favorites/${userId}`)
    const favorites = response.data.body
    dispatch(getFavorites(favorites))
  }
  const deleteFavoriteHook = async (postId) => {
    await socialApi.delete(`/favorites?postId=${postId}&userId=${userId}`)
  }
  const checkFavoritesHook = async () => {
    dispatch(checkFavorites())
  }

  return {
    getPostsToHomeHook,
    getFriendsFromFriendsHook,
    addFriendshipHook,
    likeAPost,
    makeAComment,
    getCommentsByPost,
    deleteComment,
    checkCommentsHook,
    searchUserByAlias,
    checkEmptySearchBarHook,
    sendPathHook,
    setLastUserVisitedHook,
    getInfoFromTheUserLoggedIn,
    checkInfoUserHook,
    getPostsFromUserLoggedIn,
    keepLastUserVisitesOnRedux,
    addToFavorites,
    getFavoritesHook,
    deleteFavoriteHook,
    checkFavoritesHook
  }
}
