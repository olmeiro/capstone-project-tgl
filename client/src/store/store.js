import { configureStore } from '@reduxjs/toolkit'

import { authSlice, profileSlice, homeSlice, friendSlice } from './index'

export const store = configureStore({
  reducer: {
    auth: authSlice.reducer,
    profile: profileSlice.reducer,
    home: homeSlice.reducer,
    friends: friendSlice.reducer
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false
  })
})
