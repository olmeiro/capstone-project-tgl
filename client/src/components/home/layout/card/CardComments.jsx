import React, { useState, useEffect } from 'react'

import { useHomeStore } from '../../../../hooks/useHomeStore'
import { useSelector } from 'react-redux'
import photoDefault from '../../../../../assets/photoDefault.png'
import Swal from 'sweetalert2'

export default function CardComments ({ postId }) {
  const { getCommentsByPost, deleteComment, checkCommentsHook } = useHomeStore()
  const [comments, setComments] = useState()
  const { checkComments } = useSelector(state => state.home)
  const { user } = useSelector(state => state.auth)
  const userId = user.id

  const getComments = async () => {
    const getComments = await getCommentsByPost(postId)
    setComments(getComments)
  }
  const handleDeleteComment = (commentId) => {
    Swal.fire({
      title: 'Are you sure?',
      text: "You won't be able to revert this!",
      icon: 'warning',
      showCancelButton: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
      confirmButtonText: 'Yes, delete it!'
    }).then(result => {
      if (result.isConfirmed) {
        deleteComment(commentId)
        checkCommentsHook()
        Swal.fire(
          'Deleted!',
          'Your file has been deleted.',
          'success'
        )
      }
    })
  }
  useEffect(() => {
    getComments()
  }, [checkComments])

  return (
    <div>
      <div className="border border-gray-300 shadow rounded-md p-4 max-w-sm w-full mx-auto">
        <ul role="list" className="divide-y divide-gray-200 dark:divide-gray-700">
          {
            comments && comments.map(comment => {
              return (
                <li className="py-3 sm:py-4" key={comment.id}>
                  <div className="flex items-center space-x-4">
                    <div className="flex-shrink-0">
                      <img className="w-8 h-8 rounded-full" src={comment.user.photoProfile ? comment.user.photoProfile : photoDefault} alt="image not found" />
                    </div>
                    <div className="flex-1 min-w-0">
                      <p className="text-sm font-medium text-gray-900 truncate dark:text-white">
                        {comment.user.alias}
                      </p>
                      <div className="font-normal text-gray-700 dark:text-gray-400">
                        {comment.comment}
                        <p className="text-xs">{comment.date}</p>
                      </div>
                    </div>
                    <div className="inline-flex items-center text-base font-semibold text-gray-900 dark:text-white">
                      {userId == comment.UserId
                        ? <button onClick={() => handleDeleteComment(comment.id)}>
                          <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor" strokeWidth="2">
                            <path strokeLinecap="round" strokeLinejoin="round" d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16" />
                          </svg>
                        </button>
                        : null}
                    </div>
                  </div>
                </li>
              )
            })
          }
        </ul>

        <hr />
        <br />
        <div className='flex  flex-row items-center justify-between'>
          <input type="text" placeholder='hola prueba!' />
          <svg xmlns="http://www.w3.org/2000/svg" className="h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
            <path d="M17.414 2.586a2 2 0 00-2.828 0L7 10.172V13h2.828l7.586-7.586a2 2 0 000-2.828z" />
            <path fillRule="evenodd" d="M2 6a2 2 0 012-2h4a1 1 0 010 2H4v10h10v-4a1 1 0 112 0v4a2 2 0 01-2 2H4a2 2 0 01-2-2V6z" clipRule="evenodd" />
          </svg>
        </div>
      </div>

    </div>

  )
}
