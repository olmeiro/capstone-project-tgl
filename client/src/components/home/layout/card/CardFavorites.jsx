import React, { useEffect, useState } from 'react'
import { useSelector } from 'react-redux'
import Swal from 'sweetalert2'
import { Modal } from 'flowbite-react'

import { useHomeStore } from '../../../../hooks/useHomeStore'
import photoDefault from '../../../../../assets/photoDefault.png'

export const CardFavorites = () => {
  const { deleteComment, getCommentsByPost, deleteFavoriteHook, getFavoritesHook, checkFavoritesHook, checkCommentsHook } = useHomeStore()
  const { favorites, checkFavorites } = useSelector((state) => state.home)

  const [favoriteToRender, setFavoriteToRender] = useState()
  const [openModalImg, setOpenModalImg] = useState(false)
  const [photoFavorite, setPhotoFavorite] = useState()
  const [comments, setComments] = useState()
  const { user } = useSelector((state) => state.auth)
  const userId = user.id

  const { checkComments } = useSelector((state) => state.home)
  const [idFavoriteState, setIdFavoriteState] = useState()

  const getComments = async (idFavorite) => {
    const getComments = await getCommentsByPost(idFavorite)
    setComments(getComments)
  }

  useEffect(() => {
    getComments(idFavoriteState)
  }, [checkComments])

  const handleDeleteComment = (commentId) => {
    Swal.fire({
      title: '¿estas seguro de eliminar este comentario?',
      text: 'Esta acción no se puede revertir!',
      icon: 'warning',
      showCancelButton: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
      confirmButtonText: 'Sí, eliminarlo!'
    }).then((result) => {
      if (result.isConfirmed) {
        deleteComment(commentId)
        checkCommentsHook()
        Swal.fire('Eliminado!', 'El comentario ha sido eliminado.', 'success')
      }
    })
  }

  const handleDeleteFavorite = async (postId) => {
    Swal.fire({
      title: '¿Estas seguro de eliminar este favorito?',
      text: 'Esta acción no se puede revertir!',
      icon: 'warning',
      showCancelButton: true,
      confirmButtonColor: '#3085d6',
      cancelButtonColor: '#d33',
      confirmButtonText: 'Sí, eliminarlo!'
    }).then(async (result) => {
      if (result.isConfirmed) {
        await deleteFavoriteHook(postId)
        await getFavoritesHook()
        checkFavoritesHook()
        Swal.fire('Eliminado!', 'El favorito ha sido eliminado.', 'success')
      }
    })
  }

  useEffect(() => {
    setFavoriteToRender(favorites)
  }, [favorites, checkFavorites])

  return (
    <div className="flex  flex-wrap justify-center gap-6 m-1">
      <Modal
        show={openModalImg}
        size="md"
        popup={true}
        onClose={() => setOpenModalImg(false)}
      >
        <Modal.Header />
        <Modal.Body>
          <div className="flex flex-col justify-center">
            <div className='flex justify-center mb-2'>
              <img className='rounded-md' src={photoFavorite && photoFavorite} alt="" />
            </div>

            {comments && comments.length === 0
              ? (
                <div role="status" className="flex justify-center mt-5">
                  <p>Esta publicación aún no tiene comentarios.</p>
                </div>
                )
              : (
                <div className="border border-gray-300 shadow rounded-md p-4 max-w-sm w-full mx-auto">
                <ul
                  role="list"
                  className="divide-y divide-gray-200 dark:divide-gray-700"
                >
                  {comments &&
                    comments.map((comment) => {
                      return (
                        <li className="py-3 sm:py-4" key={comment.id}>
                          <div className="flex items-center space-x-4">
                            <div className="flex-shrink-0">
                              <img
                                className="w-8 h-8 rounded-full"
                                src={
                                  comment.user.photoProfile
                                    ? comment.user.photoProfile
                                    : photoDefault
                                }
                                alt="image not found"
                              />
                            </div>
                            <div className="flex-1 min-w-0">
                              <p className="text-sm font-medium text-gray-900 truncate dark:text-white">
                                {comment.user.alias}
                              </p>
                              <div className="font-normal text-gray-700 dark:text-gray-400">
                                <div className="text-ellipsis">
                                  {comment.comment}
                                </div>
                                <p className="text-xs">{comment.date}</p>
                              </div>
                            </div>
                            <div className="inline-flex items-center text-base font-semibold text-gray-900 dark:text-white">
                              {userId === comment.UserId
                                ? (
                                <button
                                  onClick={() =>
                                    handleDeleteComment(comment.id)
                                  }
                                >
                                  <svg
                                    xmlns="http://www.w3.org/2000/svg"
                                    className="h-6 w-6"
                                    fill="none"
                                    viewBox="0 0 24 24"
                                    stroke="currentColor"
                                    strokeWidth="2"
                                  >
                                    <path
                                      strokeLinecap="round"
                                      strokeLinejoin="round"
                                      d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                                    />
                                  </svg>
                                </button>
                                  )
                                : null}
                            </div>
                          </div>
                        </li>
                      )
                    })}
                </ul>
              </div>
                )}
          </div>
        </Modal.Body>
      </Modal>
      {favoriteToRender &&
        favoriteToRender.map((favorite) => {
          const photoFavorite = favorite && favorite.photo
          const idFavorite = favorite && favorite.id

          return (
            <div key={favorite && favorite.id}>
              <div className="rounded-lg">
                <div className="bg h-72 w-72 rounded-md">
                  <img
                    className=" h-72 w-72 rounded-tr-lg rounded-tl-lg shadow-lg"
                    src={photoFavorite}
                    alt="image publication"
                  />
                </div>
                <div className="flex flex-col items-center pb-10 bg rounded-br-lg rounded-bl-lg">
                  <h5 className="w-60 h-6 mb-1 mt-2 text-sm font-semibold text-team-blue tracking-wide text-center">
                    {favorite && favorite.description}
                  </h5>
                  <span className="mt-4 text-team-green dark:text-gray-400 font-thin">
                    {favorite && favorite.date}
                  </span>
                  <div className="mt-2 flex space-x-3">
                    <button
                      href="#"
                      onClick={() => {
                        setOpenModalImg(true)
                        setPhotoFavorite(photoFavorite)
                        getComments(idFavorite)
                        setIdFavoriteState(idFavorite)
                      }}
                      className="relative -bottom-4 inline-flex items-center rounded-lg bg-team-brown py-2 px-4 text-center text-sm font-medium text-bg hover:bg-blue-800 hover:text-white focus:outline-none focus:ring-4 focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
                    >
                      Ver mas
                    </button>
                    <button
                      href="#"
                      onClick={() => handleDeleteFavorite(favorite.id)}
                      className="relative -bottom-4 inline-flex items-center rounded-lg bg-team-brown py-2 px-4 text-center text-sm font-medium text-bg hover:bg-blue-800 hover:text-white focus:outline-none focus:ring-4 focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
                    >
                      eliminar favorito
                    </button>
                  </div>
                </div>
              </div>
            </div>
          )
        })}
    </div>
  )
}
