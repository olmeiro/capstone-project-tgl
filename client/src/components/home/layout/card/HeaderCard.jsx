import React, { useEffect, useState } from 'react'
import {
  StarIcon,
  HeartIcon,
  PlusCircleIcon,
  ThumbUpIcon
} from '@heroicons/react/outline'
import { Tooltip, Button, Modal, Label, Textarea } from 'flowbite-react'
import { useHomeStore } from '../../../../hooks/useHomeStore'
import Swal from 'sweetalert2'
import { socialApi } from '../../../../api'
import { useSelector } from 'react-redux'

export default function HeaderCard({ setIsOpen, photo, description, likes, date, likeAPost, postId, userId }) {
  const [openModal, setOpenModal] = useState(false)
  const [likesRender, setLikesRender] = useState(likes)
  const [comment, setComment] = useState('')
  const { makeAComment, checkCommentsHook, addToFavorites } = useHomeStore()
  const [userOfPost, setUserOfPost] = useState()
  const { user } = useSelector(state => state.auth)

  let entireUser = socialApi.get(`/user/byid/${userId}`).then(response => response.data.body)
  useEffect(() => {
    (async () => {
      entireUser = await entireUser
      setUserOfPost(entireUser)
    })()
  }, [])

  const handleAddFavorite = () => {
    addToFavorites(postId, user.id)
    Swal.fire('¡Publicación agregada a favoritos!')
  }

  const handleLike = () => {
    likeAPost(postId)
  }
  const handleComment = (e) => {
    setComment(e.target.value)
  }
  const handleSumit = async () => {
    await makeAComment(postId, comment)
    setComment('')
    checkCommentsHook()
    Swal.fire('Comentario publicado')
    setOpenModal(false)
  }

  return (
    <div>
      <React.Fragment>
        <Modal show={openModal} size="md" popup={true} onClose={() => setOpenModal(false)}>
          <Modal.Header />
          <Modal.Body>
            <div className="space-y-6 px-6 pb-4 sm:pb-6 lg:px-8 xl:pb-8">
              <h3 className="text-xl font-medium text-gray-900 dark:text-white">
                Deja tu comentario
              </h3>
              <div>
                <div className="mb-2 block">
                  <Label
                    htmlFor="comment"
                    value="Comentario"
                  />
                </div>
                <Textarea
                  id="comment"
                  placeholder="Escribe el comentario"
                  required={true}
                  onChange={(e) => handleComment(e)}
                  value={comment}
                />
              </div>
              <div className="w-full">
                <Button onClick={() => handleSumit()}>
                  Enviar
                </Button>
              </div>
            </div>
          </Modal.Body>
        </Modal>
      </React.Fragment>

      <div className="bg-team-blue rounded-lg ">
        <div className='flex items-center rounded'>
            <img className="h-16 w-16 p-1 rounded-full" src={userOfPost && userOfPost.photoProfile} alt="" />
            <div>
              <p className='underline leading-3 text-lg antialiased font-extrabold'>{userOfPost && userOfPost.alias.toUpperCase()}</p>
              <p className='text-xs leading-5 italic mt-2'>{ date }</p>
            </div>
        </div>
        <div className='text-left ml-16 italic text tracking-wide'>
          {description}
        </div>
        <div className=" box-content max-w-md mx-auto rounded-br-lg rounded-bl-lg  shadow-md overflow-hidden md:max-w-2xl mt-2">
          <div className=' flex justify-center'>
            <img
              className="sm:h-48 sm:w-full object-cover md:h-80 md:w-full "
              src={photo}
              alt="Man looking at item at a store"
            />
          </div>
          <header className="flex items-center p-4">
            <button
              className="bg-team-dark w-1/3 text-team-green sm:text-sm sm:mr-2 rounded px-4 py-2 hover:text-white"
              onClick={() => setIsOpen(true)}
            >
              Ver comentarios
            </button>
            <div className="flex flex-row w-2/3 justify-end items-center gap-4">
              <Tooltip content="likes" arrow={false}>
                <div>
                  {likesRender}
                </div>
              </Tooltip>
              <Tooltip content="like" arrow={false}>
                <button onClick={() => handleLike()}>
                  <ThumbUpIcon
                    className="h-6 w-6 mb-3 relative top-1  rounded-md  hover:fill-black"
                    aria-hidden="true"
                  />
                </button>
              </Tooltip>
              <Tooltip content="guardar" arrow={false}>
                <button onClick={() => handleAddFavorite()}>
                  <StarIcon
                    className="h-6 w-6 mb-3 relative top-1  rounded-md hover:fill-black"
                    aria-hidden="true"
                  />
                </button>
              </Tooltip>
              <Tooltip content="comentar" arrow={false}>
                <button>
                  <PlusCircleIcon
                    onClick={() => setOpenModal(true)}
                    className="h-6 w-6 mb-3 relative top-1  rounded-md  hover:fill-black hover:text-white"
                    aria-hidden="true"
                  />
                </button>
              </Tooltip>
            </div>
          </header>
        </div>
      </div>
    </div>
  )
}
