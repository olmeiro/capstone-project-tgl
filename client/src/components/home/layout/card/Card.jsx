import React, { useState } from 'react'
import PropTypes from 'prop-types'

import CarDrawer from './CardDrawer'
import CardComments from './CardComments'
import HeaderCard from './HeaderCard'

export const Card = ({ photo, description, likes, date, postId, likeAPost, userId }) => {
  const [isOpen, setIsOpen] = useState(false)

  return (
    <div className="mb-3">
      <HeaderCard
        setIsOpen={setIsOpen}
        userId={userId}
        photo={photo}
        description={description}
        likes={likes}
        date={date}
        likeAPost={likeAPost}
        postId={postId}
      />
      <CarDrawer isOpen={isOpen} setIsOpen={setIsOpen}>
        <CardComments postId={postId} />
      </CarDrawer>
    </div>
  )
}

Card.propTypes = {
  photo: PropTypes.string,
  description: PropTypes.string,
  likes: PropTypes.number,
  date: PropTypes.string,
  postId: PropTypes.number,
  likeAPost: PropTypes.func,
  userId: PropTypes.number
}
