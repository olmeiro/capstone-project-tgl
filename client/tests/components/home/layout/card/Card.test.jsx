import { Card } from '../../../../../src/components/home/layout/card/Card'
import { render, screen } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider, useSelector } from 'react-redux'
import { homeSlice } from '../../../../../src/store'
import { useHomeStore } from '../../../../../src/hooks/useHomeStore'

const store = configureStore({
  reducer: {
    home: homeSlice.reducer
  }
})

jest.mock('../../../../../src/hooks/useHomeStore')
jest.mock('../../../../../src/components/home/layout/card/Card')

jest.mock('react-redux', () => ({
  useSelector: jest.fn()
}))

describe('tests component <Card />', () => {
  test('should render component', () => {
    useSelector.mockImplementation(() => ('hello world'))
    useHomeStore.mockReturnValue({
      useDispatch: jest.fn()
    })

    const { container } = render(
      // <Provider store={store}>
        <Card />
      // </Provider>
    )
    screen.debug()
    expect(container).toMatchSnapshot()
  })
})
