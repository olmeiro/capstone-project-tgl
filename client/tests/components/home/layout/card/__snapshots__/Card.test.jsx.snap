// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests component <Card /> should render component 1`] = `<div />`;
